libpod/oci_conmon_exec_linux.go

```diff
@@ -613,19 +613,38 @@ func attachExecHTTP(c *Container, sessionID string, r *http.Request, w http.Resp
 			logrus.Debugf("Performing non-terminal HTTP attach for container %s", c.ID())
 			err = httpAttachNonTerminalCopy(conn, httpBuf, c.ID(), attachStdin, attachStdout, attachStderr)
 		}
-		errChan <- err
+		stdoutChan <- err
 		logrus.Debugf("STDOUT/ERR copy completed")
 	}()
 
-	if cancel != nil {
+	inDone := false
+	if !attachStdin {
+		inDone = true
+	}
+	outDone := false
+
+	for {
 		select {
-		case err := <-errChan:
-			return err
+		case err := <-stdoutChan:
+			if err != nil {
+				return err
+			}
+
+			outDone = true
+			if outDone && inDone {
+				return nil
+			}
+		case err := <-stdinChan:
+			if err != nil {
+				return err
+			}
+
+			inDone = true
+			if outDone && inDone {
```

Comment:
Nit: Putting these three lines at the bottom would make this cleaner and not have duplicated code.